muon: add

hovercats opened the pull request:
package muon, as mentioned on irc.
we also need libpkgconf for muon to be able to find dependencies.
Ive tried without it, and finding dependencies becomes a nightmare, atleast it did for me. maybe you know something I dont, if so, im all ears.

Im also somewhat dissatisfied with having to include `embedded_files.h`.
Ill look more into if we can build it instead of including it.
If you have a solution for this, let me know.

rnpnr commented:
> we also need libpkgconf for muon to be able to find dependencies.

muon has a bundled libpkgconfig. Since its only needed for muon its probably fine to just use that one (I think other packages in oasis do this as well). Michael can confirm though.

Also does muon not need a pkg-config binary on the system to find stuff? I would assume your system has one so you probably wouldn't have noticed but it is listed in the [dependencies](https://git.sr.ht/~lattis/muon#dependencies).

hovercats commented:
> Also does muon not need a pkg-config binary on the system to find stuff? I would assume your system has one so you probably wouldn't have noticed but it is listed in the dependencies.

I did some testing here, to see whats what again, and this seems to only be partially true.
Same as you, I also use `u-config`.

I removed `u-config`, and did a rebuild of all my packages currently using `muon`.
And so far, only `gtk+3` seemed to mess this up for me. hovercats/oakiss#2 for details.

Reinstalling `u-config`, I removed support for libpkgconf for `muon`, and redid my test.
This time it was much more hit or miss. Some packages built just fine, whereas others complained about not finding dependencies.
Which is kinda what I expect.

If my memory serves me right, I opted to not use libpkgconf support when I packaged it using oasis, because why include it if you can manage without?
However, I did not have great success in doing so, which sortof forced me to include support for it.
This was also with a pkgconf implementation installed.

Im only left with one conclusion here, and that is that we need support for libpkgconf, atleast for now.
Muon is still in an early stage, and lots of packages doesnt build with muon, for all kinds of reasons.
Adding not being able to find dependencies because we wanna skimp out on one measly feature doesnt make sense to me.

In any case, Im not oposed to building pkgconf binary (or preferably u-config, which is less than half of pkgconf's size), and add to this pr, or a separate one. depending on what @mcf wants.
